x-pack/plugins/task_manager/server/queries/query_clauses.ts

```diff
@@ -39,52 +160,99 @@ export interface SortClause {
     };
   };
 }
+export type SortOptions = string | SortClause | Array<string | SortClause>;
+
 export interface ScriptClause {
   source: string;
   lang: string;
   params: {
     [field: string]: string | number | Date;
   };
 }
-export interface UpdateByQuery<T> {
-  query: BoolClause<T>;
-  sort: SortClause;
+
+export interface UpdateByQuery<T extends BoolClauseFilter> {
+  query: PinnedQuery<T> | BoolClauseWithAnyCondition<T>;
+  sort: SortOptions;
   seq_no_primary_term: true;
   script: ScriptClause;
 }
 
-export function shouldBeOneOf<T>(
-  ...should: Array<BoolClause<T> | IDsClause | T>
-): {
-  bool: ShouldClause<T>;
-} {
+export interface PinnedQuery<T extends BoolClauseFilter> {
+  pinned: PinnedClause<T>;
+}
+
+export interface PinnedClause<T extends BoolClauseFilter> {
+  ids: string[];
+  organic: BoolClauseWithAnyCondition<T>;
+}
+
+export function mergeBoolClauses<T extends BoolClauseFilter>(
```

Comment:
Doing a straight merge on queries that use a combination of `should` with `must` or `must_not` can be dangerous if that's the case here.

Reply:
Same would apply with merging `should` queries as I believe they are AND together instead of combined into a single OR.

Reply:
My thinking is that if someone is explicitly merging them, they mean for them to be applied as an AND, as it only merges the root object.
If someone wants an `OR` they should use the `shouldBeOneOf`.

We could rename `mergeBoolClauses` to something that expresses this better, but not sure what it should be 🤔

Reply:
upon further inspection - this does merge deeply, my mistake, I'll simplify this.